src/pages/sighting/encounters/Encounters.jsx

```diff
@@ -85,26 +97,59 @@ export default function Encounters({
   const encounterFieldSchemas = useEncounterFieldSchemas();
   const encounters = get(sightingData, 'encounters', []);
 
+  useEffect(
+    () => {
+      setMessageForConfirmDelete(
+        'CONFIRM_DELETE_ENCOUNTER_DESCRIPTION',
+      );
+      if (vulnerableObject) {
+        setMessageForConfirmDelete('BOTH_VULNERABLE_MESSAGE');
+      }
+    },
+    [vulnerableObject],
+  );
+
   return (
     <div>
       <ConfirmDelete
         open={Boolean(encounterToDelete)}
         onClose={() => setEncounterToDelete(null)}
         onDelete={async () => {
           let successful;
+          const vulnObjKeys = vulnerableObject
+            ? Object.keys(vulnerableObject)
+            : [];
+          const vulnerableObjStr = get(vulnObjKeys, '0');
           if (pending) {
             successful = await deleteAGSEncounter(
               sightingId,
               encounterToDelete,
             );
-          } else {
+          }
+          if (!pending && vulnerableObjStr) {
+            successful = await deleteSightingEncounter(
+              encounterToDelete,
+              true,
+              true,
+            );
+          }
+          if (!pending && !vulnerableObjStr) {
             successful = await deleteSightingEncounter(
               encounterToDelete,
             );
           }
           if (successful) {
             setEncounterToDelete(null);
-            refreshSightingData();
+            const navigateAway =
+              Boolean(vulnerableObject) && encounters.length <= 1;
```

Comment:
you don't need to cast to boolean in an expression (it happens automatically). so you can just write 
```
const navigateAway = vulnerableObject && encounters.length <= 1;
```

Reply:
Good point. Done.